src/test/unit/types/SdrTest.cpp

```diff
@@ -139,61 +139,48 @@ TEST(SdrTest, TestSDR_Examples) {
 
 TEST(SdrTest, TestSetDenseVec) {
     SDR a({11, 10, 4});
-    Byte *before = a.getDense().data();
-    SDR_dense_t vec = vector<Byte>(440, 1);
-    Byte *data = vec.data();
+    auto before = a.getDense().data();
+    SDR_dense_t vec = SDR_dense_t(440, 1);
+    auto data = vec.data();
     a.setDense( vec );
-    Byte *after = a.getDense().data();
+    auto after = a.getDense().data();
     ASSERT_NE( before, after ); // not a copy.
     ASSERT_EQ( after, data ); // correct data buffer.
 }
 
-TEST(SdrTest, TestSetDenseByte) {
-    SDR a({11, 10, 4});
-    auto vec = vector<Byte>(a.size, 1);
-    a.zero();
-    a.setDense( (Byte*) vec.data());
-    ASSERT_EQ( a.getDense(), vec );
-    ASSERT_NE( ((vector<Byte>&) a.getDense()).data(), vec.data() ); // true copy not a reference
-    ASSERT_EQ( a.getDense().data(), a.getDense().data() ); // But not a copy every time.
-}
-
 TEST(SdrTest, TestSetDenseUInt) {
     SDR a({11, 10, 4});
-    auto vec = vector<UInt>(a.size, 1);
-    a.setDense( (UInt*) vec.data() );
-    ASSERT_EQ( a.getDense(), vector<Byte>(a.size, 1) );
+    auto vec = SDR_dense_t(a.size, 1);
+    a.setDense(vec );
+    ASSERT_EQ( a.getDense(), SDR_dense_t(a.size, 1) );
     ASSERT_NE( a.getDense().data(), (const Byte*) vec.data()); // true copy not a reference
 }
 
 TEST(SdrTest, TestSetDenseInplace) {
     SDR a({10, 10});
     auto& a_data = a.getDense();
-    ASSERT_EQ( a_data, vector<Byte>(100, 0) );
+    ASSERT_EQ( a_data, SDR_dense_t(100, 0) );
     a_data.assign( a.size, 1 );
     a.setDense( a_data );
     ASSERT_EQ( a.getDense().data(), a.getDense().data() );
     ASSERT_EQ( a.getDense().data(), a_data.data() );
-    ASSERT_EQ( a.getDense(), vector<Byte>(a.size, 1) );
+    ASSERT_EQ( a.getDense(), SDR_dense_t(a.size, 1) );
     ASSERT_EQ( a.getDense(), a_data );
 }
 
 TEST(SdrTest, TestSetSparseVec) {
     SDR a({11, 10, 4});
-    UInt *before = a.getSparse().data();
-    auto vec = vector<UInt>(a.size, 1);
-    UInt *data = vec.data();
-    for(UInt i = 0; i < a.size; i++)
-        vec[i] = i;
-    a.setSparse( vec );
-    UInt *after = a.getSparse().data();
+    const auto before = a.getSparse();
+    const SDR_sparse_t data = {1, 2, 3};
```

Comment:
This testcase doesn't do what it did before, you've changed the substance of it.  It doesn't call the same method in SDR (different overloads for non/const input).

Reply:
thanks for noticing! I'll try to bring it back in its orig meaning